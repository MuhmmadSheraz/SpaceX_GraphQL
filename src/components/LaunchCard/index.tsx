import React, { useEffect } from "react";
import { UpcomingLaunchesListQuery } from "../../generated/graphql";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";

import Typography from "@material-ui/core/Typography";
const useStyles = makeStyles((theme) => ({
  rootcard: {
    maxWidth: 450,
    maxHeight: 1000,
  },
  media: {
    // height: 140,
  },
  root: {
    flexGrow: 1,
  },
  paper: {
    padding: theme.spacing(2),
    textAlign: "center",
    color: theme.palette.text.secondary,
  },
}));
type card = {
  data: any;
};
const LaunchCard: React.FC<card> = ({ data }) => {
  const classes = useStyles();

  const goto = (a: string) => {
    window.open(a);
  };
  return (
    <Card className={classes.rootcard}>
      <CardActionArea>
        <CardMedia className={classes.media} title="Contemplative Reptile" />
        <img
          src={
            data && data.links.flickr_images[0]
              ? data.links.flickr_images[0]
              : "https://upload.wikimedia.org/wikipedia/commons/thumb/a/ac/No_image_available.svg/600px-No_image_available.svg.png"
          }
          alt="Missile"
          style={{
            maxWidth: "100%",
            maxHeight: "30vh",
            minHeight: "50vh",
            minWidth: "100%",
          }}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {data?.mission_name}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            Launch Year :{data?.launch_year}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Link to={`/MissionDetails/${data && data?.flight_number}`}>
          <Button size="small" color="primary">
            Details
          </Button>
        </Link>
        <Button
          size="small"
          color="primary"
          onClick={() =>
            goto(
              data && data.links && data.links.wikipedia
                ? data.links.wikipedia
                : ""
            )
          }
        >
          Learn More
        </Button>
      </CardActions>
    </Card>
  );
};
export default LaunchCard;
