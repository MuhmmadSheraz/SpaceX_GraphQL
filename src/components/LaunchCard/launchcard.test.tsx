import React from "react";
import LaunchCard from "./index";

import { shallow } from "enzyme";
describe("App Component", () => {
  it("Should Passes UI Test", () => {
    const wrapper = shallow(<LaunchCard data="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
