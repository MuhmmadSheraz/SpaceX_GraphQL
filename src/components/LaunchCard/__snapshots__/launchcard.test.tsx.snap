// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should Passes UI Test 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-rootcard-1"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-2"
      title="Contemplative Reptile"
    />
    <img
      alt="Missile"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/ac/No_image_available.svg/600px-No_image_available.svg.png"
      style={
        Object {
          "maxHeight": "30vh",
          "maxWidth": "100%",
          "minHeight": "50vh",
          "minWidth": "100%",
        }
      }
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        gutterBottom={true}
        variant="h5"
      />
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body2"
      >
        Launch Year :
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))>
    <Link
      to="/MissionDetails/"
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        size="small"
      >
        Details
      </WithStyles(ForwardRef(Button))>
    </Link>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
      size="small"
    >
      Learn More
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
